### PR TITLE
PT-3346: Filter panel should remain collapsed until user either opens it or logs out of PhenoTips

### PR DESCRIPTION
**Pull request by marta-**

### CODE REVIEW

File: components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml [hunk 1]
```diff
@@ -251,25 +251,44 @@ document.observe('xwiki:livetable:newrow', function(tableEvent){
...
+    var toggleFilters = function (filtersElt, forceHide) {
+      if (filtersElt) {
+        filtersElt.toggleClassName('collapsed', forceHide);
+        filtersElt.up('.xwiki-livetable-container').toggleClassName('hidden-filters', forceHide);
+        var key = getCookieKey(filtersElt.up('.entity-directory').id);
+        if (filtersElt.hasClassName('collapsed')) {
+          XWiki.cookies.create(key, 'hidden', '')
+        } else {
+          XWiki.cookies.erase(key)
+       }
```
Comment (veronikaslc) [inline components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml]: couple of very minor comment:  indentation is missing one space; eye got used to `;` at the end of a line although they aren't necessary in those `if-else` one lines

Comment (veronikaslc) [inline components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml]: other than that looks fine to me

Comment (marta-) [inline components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml]: @veronikaslc thanks, fixed.

File: components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml [hunk 2]
```diff
@@ -251,25 +251,44 @@ document.observe('xwiki:livetable:newrow', function(tableEvent){
       </use>
     </class>
     <property>
-      <cache>long</cache>
+      <cache>forbid</cache>
```
Comment (sdumitriu) [inline components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml]: I don't like this.

File: components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml [hunk 3]
```diff
@@ -251,25 +251,44 @@ document.observe('xwiki:livetable:newrow', function(tableEvent){
...
-    if (filtersElt) {
-      filtersElt.toggleClassName('collapsed');
-      filtersElt.up('.xwiki-livetable-container').toggleClassName('hidden-filters');
-    }
-  });
-});</code>
+      <code>(function() {
+  var init = function(event) {
+    var getCookieKey = function (tableId) {
+      return "$!{xcontext.userReference.hashCode()}" + '_' + tableId + '_filters_state';
```
Comment (sdumitriu) [inline components/navigation/ui/src/main/resources/PhenoTips/DBWebHomeSheet.xml]: Better place the user name somewhere in the HTML.

**Comment by marta-** [comment]
@sdumitriu comments addressed.

@sashaandjic @veronikaslc this needs to be reviewed and tested again. Thanks.

**Comment by veronikaslc** [comment]
tested, works

**Comment by sashaandjic** [comment]
Behaviour changed a bit...Previously we had 
Filter panels should on browse all records pages were be open by default when user logs in. Once a specific filter panel is collapsed, it remained collapsed unless:
1. user opens it again
2. user logs out and logs in again
Now #2 is no longer valid.

**Comment by marta-** [comment]
Thanks @sashaandjic , it was a bug and I committed the fix.

**Comment by sashaandjic** [comment]
There is still discrepancy between present behaviour and JIRA.
Currently the state of filters is saved per user, and whenever user logs in the filters will look the same as in the last session.
In JIRA rask it was specified that every time someone logs in, the filters should be open.

**Comment by marta-** [comment]
@sashaandjic That was by design. From my experience it makes more sense to find the filters the way you left them when you log back in, than to have everything reset to defaults. I should have changed the jira task to reflect this.

**Comment by sashaandjic** [comment]
Ok. I will adjust task description in JIRA. Thank you.

**Pull request merged**